[17.0][FW] fieldservice: multiple ports from 16.0

brian10048 opened the pull request:
Port from 16.0 to 17.0:
- #1162
- #1163
- #1164
- #1165
- #1194
- #1202
- #1221
- #1286
- #1304
- #1317
- #1346

brian10048 committed:
[IMP] fieldservice: fsm_category

compute method should depend on fields used in the calculation

brian10048 committed:
[IMP] fieldservice

Improve fsm.category parent/child relationship and optimize for search and views

HekkiMelody committed:
[IMP]fieldservice: stage names translatable

victoralmau committed:
[IMP] fieldservice: Add freeze_time to tests to avoid false errors sometimes.

victoralmau committed:
[IMP] fieldservice: Add TestFSMOrderBase

victoralmau committed:
[FIX] fieldservice: Add recursive=True to full_name field to prevent warning log

houzefa-abba committed:
[FIX] fieldservice: Fix HTML field declaration

Type mix-up was provoking infinite loops when navigating out of form
views with "next" arrow buttons.

Fixes #1161.

houzefa-abba committed:
[FIX] fieldservice: Fix manual location creation

Read-only+required partner field shown in debug mode was preventing
manual creation in debug mode or with OCA module base_technical_features.

houzefa-abba committed:
[FIX] fieldservice: Fix location partner type

Service locations `fsm.location` create a partner `res.partner` through
inheritance.

Before this change, partners could be explicitely typed "fsm_location"
when adding them within children of an existing partner, which in turn
would automatically create service locations on top of them.

But creating a service location by hand would not set that type on the
partner created through inheritance. After this change, all partners
built from service locations are typed the same.

Fixes #1128.

houzefa-abba committed:
[IMP] fieldservice: Set location parent to owner

Before this change, service location partners (created through
inheritance) were typed "fsm_location" but did not always have a parent.

When adding them within children of an existing partner, they of course
did have a parent, but not the other way around: they had no parents
when created, through inheritance, upon service location creation.

After this change, service location partners are by default children of
the service location owner partner.

This is only done through defaults but not as related fields, which
would break too much. Therefore, existing partners are not impacted by
this change; only new ones created when creating a service location.

See #1128.

brian10048 committed:
[FIX] fieldservice: schedule_date_start bug

Update scheduled_date_start field not date_start field

brian10048 committed:
[FIX] fieldservice: assign correct date_end

pcastelovigo committed:
allow to change partner

 #1166 also in 16.0

brian10048 committed:
[FIX] fieldservice: order check for holiday

ppyczko committed:
[IMP] fieldservice: Display FSM Location address on FSM order form view

HekkiMelody committed:
[FIX] fieldservice: unaccent parent path

Silences the following warning:

WARNING ? odoo.models: parent_path field on model 'fsm.category'
should have unaccent disabled. Add `unaccent=False`
to the field definition.

rvalyi reviewed (approved):
(code review only)

brian10048 commented:
@victoralmau whenever I introduce this commit a425291149e73bd69d8d51228c1f6496cd9f7d1f the tests hang up on test_fsm_order.py . I'm not too familiar with freeze_time, is there something that should be done differently?

victoralmau commented:
> @victoralmau whenever I introduce this commit [a425291](https://github.com/OCA/field-service/commit/a425291149e73bd69d8d51228c1f6496cd9f7d1f) the tests hang up on test_fsm_order.py . I'm not too familiar with freeze_time, is there something that should be done differently?

Ok, I see, the way to solve it is the following:
- Remove `freeze_time` from class `TestFSMOrder`.
- Add `freeze_time` in the tests that require it.

Actually this “problem” is not explained anywhere, but I have discovered it recently and it happens when tests are run post-install.

brian10048 commented:
@victoralmau that was it! Thank you!

victoralmau committed:
[IMP] fieldservice: Define @freeze_time decorator in tests and not in TestFSMOrderBase to avoid side effects

If a module has a test that extends it, a crash occurs in post tests

Related to https://github.com/OCA/field-service/pull/1340#issuecomment-2711247458

victoralmau commented:
Already fixed in v16 https://github.com/OCA/field-service/pull/1346

victoralmau reviewed (approved):
Code review OK.

brian10048 commented:
/ocabot merge major
